### PR TITLE
[Backport v3.6-branch] drivers: mipi_dbi: mipi_dbi_spi: do not take spinlock

### PR DESCRIPTION
**Pull request by zephyrbot**
Backport 5b6fadc10d3be7120a808928f11389a6151944d2 from #69988.

Fixes: https://github.com/zephyrproject-rtos/zephyr/issues/68815

### CODE REVIEW

**Review by henrikbrixandersen — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]
Feels a bit strange to approve this given I wrote it, but I think I need to since I'm the assignee

**Comment by henrikbrixandersen** [comment]
> Feels a bit strange to approve this given I wrote it, but I think I need to since I'm the assignee

Right. As the maintainer, you approve that this is something that should be backported. Anyone could have flagged it for backporting, so we need maintainer approval. 

When a maintainer submits patches for the subsystems, they maintain, to the main branch, their approval is implicit.

**Pull request merged**
